src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,485 @@
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path);
+                watcher.Filters.Add("*.pdb");
+                watcher.Filters.Add("*.dll");
+
+                watcher.Filters.Clear();
+                Assert.Equal("*", watcher.Filter);
+                Assert.Equal(new string[] { }, watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void GetFiltersAfterFiltersClear()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path);
+                watcher.Filters.Add("*.pdb");
+                watcher.Filters.Add("*.dll");
+
+                watcher.Filters.Clear();
+                Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters[0]);
+                Assert.Equal(0, watcher.Filters.Count);
+                Assert.Empty(watcher.Filters);
+                Assert.NotNull(watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void InvalidOperationsOnFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            watcher.Filters.Add("*.pdb");
+            watcher.Filters.Add("*.dll");
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters.Insert(4, "*"));            
+            watcher.Filters.Clear();
+            Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters[0]);
+        }
+
+        [Fact]
+        public void SetAndGetFilterProperty()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path, "*.pdb");
+                watcher.Filters.Add("foo");
+                Assert.Equal(2, watcher.Filters.Count);
+                Assert.Equal(new string[] { "*.pdb", "foo" }, watcher.Filters);
+
+                watcher.Filter = "*.doc";
+                Assert.Equal(1, watcher.Filters.Count);
+                Assert.Equal("*.doc", watcher.Filter);
+                Assert.Equal("*.doc", watcher.Filters[0]);
+                Assert.Equal(new string[] { "*.doc" }, watcher.Filters);
+
+                watcher.Filters.Clear();
+                Assert.Equal("*", watcher.Filter);
+            }
+        }
+
+        [Fact]
+        public void SetAndGetFiltersProperty()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path, "*.pdb");
+                watcher.Filters.Add("foo");
+                Assert.Equal(new string[] { "*.pdb", "foo" }, watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Delete()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
```

Comment:
You already derive from FileCleanupTestBase, no need to use TempDirectory directly.  I've rewritten this test for clarity, hopefully this will help:

``` C#
[Fact]
public void FileSystemWatcher_File_Delete_MultipleFilters()
{
    // Check delete events against multiple filters

    DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
    FileInfo fileOne = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
    FileInfo fileTwo = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
    fileOne.Create().Dispose();
    fileTwo.Create().Dispose();

    using (var watcher = new FileSystemWatcher(directory.FullName))
    {
        watcher.Filters.Add(fileOne.Name);
        watcher.Filters.Add(fileTwo.Name);

        ExpectEvent(watcher, WatcherChangeTypes.Deleted, () => fileOne.Delete(), cleanup: null, expectedPath: fileOne.FullName);
        ExpectEvent(watcher, WatcherChangeTypes.Deleted, () => fileTwo.Delete(), cleanup: null, expectedPath: fileTwo.FullName);
    }
}
```

In the rewrite, I've grouped things together into phases. Setup (1), then execution (2).  In the original code setup happens interleaved with the execution. It becomes a mental exercise to understand if the setup has to be that way- is there an order dependency? In the update above it is clear there isn't.